driver/pgdriver/config.go

```diff
@@ -173,6 +176,12 @@ func WithWriteTimeout(writeTimeout time.Duration) Option {
 	}
 }
 
+func WithResetSessionFunc(fn func(context.Context, *Conn) error) Option {
```

Comment:
Could you add a brief comment explaining what this is useful for you and potentially the rest of us?